src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -0,0 +1,335 @@
+        this.flowFrameworkIndicesHandler = flowFrameworkIndicesHandler;
+        this.encryptorUtils = encryptorUtils;
+    }
+
+    @Override
+    protected void doExecute(Task task, WorkflowRequest request, ActionListener<WorkflowResponse> listener) {
+        // Retrieve use case template from global context
+        String workflowId = request.getWorkflowId();
+        GetRequest getRequest = new GetRequest(GLOBAL_CONTEXT_INDEX, workflowId);
+
+        // Stash thread context to interact with system index
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.wrap(response -> {
+                context.restore();
+
+                if (!response.isExists()) {
+                    listener.onFailure(
+                        new FlowFrameworkException(
+                            "Failed to retrieve template (" + workflowId + ") from global context.",
+                            RestStatus.NOT_FOUND
+                        )
+                    );
+                    return;
+                }
+
+                // Parse template from document source
+                Template template = Template.parse(response.getSourceAsString());
+
+                // Decrypt template
+                template = encryptorUtils.decryptTemplateCredentials(template);
+
+                // Sort and validate graph
+                Workflow provisionWorkflow = template.workflows().get(PROVISION_WORKFLOW);
+                List<ProcessNode> provisionProcessSequence = workflowProcessSorter.sortProcessNodes(provisionWorkflow, workflowId);
+                workflowProcessSorter.validateGraph(provisionProcessSequence);
+
+                // We have a valid template and sorted nodes, get the created resources
+                getResourcesAndExecute(request.getWorkflowId(), provisionProcessSequence, listener);
+            }, exception -> {
+                if (exception instanceof FlowFrameworkException) {
+                    logger.error("Workflow validation failed for workflow : " + workflowId);
+                    listener.onFailure(exception);
+                } else {
+                    logger.error("Failed to retrieve template from global context.", exception);
+                    listener.onFailure(new FlowFrameworkException(exception.getMessage(), ExceptionsHelper.status(exception)));
+                }
+            }));
+        } catch (Exception e) {
+            logger.error("Failed to retrieve template from global context.", e);
+            listener.onFailure(new FlowFrameworkException(e.getMessage(), ExceptionsHelper.status(e)));
+        }
+    }
+
+    private void getResourcesAndExecute(
+        String workflowId,
+        List<ProcessNode> provisionProcessSequence,
+        ActionListener<WorkflowResponse> listener
+    ) {
+        GetWorkflowRequest getRequest = new GetWorkflowRequest(workflowId, true);
```

Comment:
With PR #273 merged in, this line will have to change to `GetWorkflowStateRequest(workflowId, true)`